File: ActivityStack/app/src/main/java/com/example/velm/activitystack/MainActivity.java
package com.example.velm.activitystack;

import android.app.ActivityManager;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import java.util.List;

public class MainActivity extends AppCompatActivity {


    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button = (Button)findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(),SecondActivity.class);
                intent.setFlags( Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);
            }
        });
        ActivityManager activityManager = (ActivityManager) getSystemService(ACTIVITY_SERVICE);

        List<ActivityManager.RunningTaskInfo> tasklist = activityManager.getRunningTasks(10);

        for (int i =0; i < tasklist.size();i++){
            Log.d("TASK",i+" = "+tasklist.get(i).topActivity.getClassName());
        }
    }
}

File: InstrumentTestingExample/app/src/main/java/com/subaru/instrumenttestingexample/MainActivity.kt
package com.subaru.instrumenttestingexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        helloworld.setOnClickListener {
            clickListener()
        }
    }

    private fun clickListener() {
        println("Helllp")
    }
}